### PR TITLE
refactor: migrate DialogContentContainer to typescript

### PR DESCRIPTION
**Pull request by vishal-codes**
Signed-off-by: vishal-codes <itsvishal2417@gmail.com>

#### Basic
- [ ] Used plop (`npm run plop`) to create a new component.
- [x] PR has description.
- [ ] New component is functional and uses Hooks. 
- [ ] Component defines [`PropTypes`](https://reactjs.org/docs/typechecking-with-proptypes.html).
#### Style
- [ ] Styles are added to `NewComponent.modules.scss` file inside of the `NewComponent` folder.
- [ ] Component uses CSS Modules.
- [ ] Design is compatible with [Monday Design System](https://design.monday.com/).
#### Storybook
- [ ] Stories were added to `/src/NewComponent/__stories__/NewComponent.stories.js` file.
- [ ] Stories include all flows of using the component.
#### Tests
- [ ] Tests are compliant with [TESTING_README.md](TESTING_README.md) instructions.
#### Description
This P.R. migrates "DialogContentContainer" and it's tests to typescript.
this resolves https://github.com/mondaycom/monday-ui-react-core/issues/900

### CODE REVIEW

File: src/components/DialogContentContainer/DialogContentContainer.tsx [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
+  children: React.ReactNode;
+  className: string;
+  ariaLabelledby?: string;
+  ariaDescribedby?: string;
+  type: "modal" | "popover";
+  size: "small" | "medium" | "large";
+  style: React.CSSProperties;
+}
+
 const DIALOG_TYPES = {
```
Comment (hadasfa) [inline src/components/DialogContentContainer/DialogContentContainer.tsx:19]: Please create new file which called DialogContentContainerConstants.ts which will contain the enums of the DialogContentContainer component.
Please create an enum for the following: DIALOG_TYPES => DialogType, DIALOG_SIZES => DialogSize.
And replace the values of types and sizes with the following.
Also, use these enums for the type and size default props

**Review by hadasfa — CHANGES_REQUESTED** [review]
Thank you so much for contributing! Please read the following comments.

File: src/components/DialogContentContainer/__tests__/dialogContentContainer.jest.tsx [hunk 2]
```diff
@@ -3,6 +3,6 @@ import renderer from "react-test-renderer";
 import DialogContentContainer from "../DialogContentContainer";
 
 it("renders correctly with empty props", () => {
-  const tree = renderer.create(<DialogContentContainer />).toJSON();
+  const tree = renderer.create(<DialogContentContainer className="testClassName" type="popover" size="medium" style={{textAlign:'center'}}><p>Children text</p></DialogContentContainer>).toJSON();
```
Comment (hadasfa) [inline src/components/DialogContentContainer/__tests__/dialogContentContainer.jest.tsx:6]: please revert the changes in the structure on the component props in the tests

File: src/components/DialogContentContainer/DialogContentContainer.tsx [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
-import PropTypes from "prop-types";
 import cx from "classnames";
 import useMergeRefs from "../../hooks/useMergeRefs";
 import { BEMClass } from "../../helpers/bem-helper";
-
+import VibeComponentProps from "src/types/VibeComponentProps";
 import "./DialogContentContainer.scss";
 
+interface DialogContentContainerProps extends VibeComponentProps {
+  children: React.ReactNode;
```
Comment (hadasfa) [inline src/components/DialogContentContainer/DialogContentContainer.tsx:10]: please mark all the props which are not required as optional

Comment (vishal-codes) [reply src/components/DialogContentContainer/DialogContentContainer.tsx:10]: I have one doubt...Is children prop optional?

File: src/components/DialogContentContainer/DialogContentContainer.tsx [hunk 4]
```diff
@@ -21,7 +30,7 @@ const DIALOG_SIZES = {
 
 const bemHelper = BEMClass("dialog-content-container");
 
-const DialogContentContainer = forwardRef(
+const DialogContentContainer: React.ForwardRefExoticComponent<DialogContentContainerProps> = forwardRef(
```
[the remark this answers is not in the document]
Comment (vishal-codes) [reply src/components/DialogContentContainer/DialogContentContainer.tsx]: Thank you for providing an example. I've fixed it.

**Review by hadasfa — APPROVED** [review]
